pkg/segment/query/processor/inputlookupcommand.go

```diff
@@ -0,0 +1,175 @@
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package processor
+
+import (
+	"compress/gzip"
+	"encoding/csv"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/siglens/siglens/pkg/config"
+	"github.com/siglens/siglens/pkg/segment/query/iqr"
+	"github.com/siglens/siglens/pkg/segment/structs"
+	"github.com/siglens/siglens/pkg/segment/utils"
+	putils "github.com/siglens/siglens/pkg/utils"
+)
+
+type inputlookupProcessor struct {
+	options   *structs.InputLookup
+	eof       bool
+	qid       uint64
+	start     uint64
+	processed uint64
+}
+
+func checkCSVFormat(filename string) bool {
+	return strings.HasSuffix(filename, ".csv") || strings.HasSuffix(filename, ".csv.gz")
+}
+
+func createRecord(columnNames []string, record []string) (map[string]utils.CValueEnclosure, error) {
+	if len(columnNames) != len(record) {
+		return nil, fmt.Errorf("createRecord: columnNames and record lengths are not equal, len(columnNames): %v, len(record): %v",
+			len(columnNames), len(record))
+	}
+	recordMap := make(map[string]utils.CValueEnclosure)
+	for i, cname := range columnNames {
+		recordMap[cname] = utils.CValueEnclosure{Dtype: utils.SS_DT_STRING, CVal: record[i]}
+	}
+	return recordMap, nil
+}
+
+func (p *inputlookupProcessor) Process(inpIqr *iqr.IQR) (*iqr.IQR, error) {
+	if inpIqr != nil && !p.options.FirstCommand {
+		p.qid = inpIqr.GetQid()
+		return inpIqr, nil
+	}
+	if p.eof {
+		return nil, io.EOF
+	}
+
+	if p.options == nil {
+		return nil, fmt.Errorf("inputlookupProcessor.Process: InputLookup is nil")
+	}
+	filename := p.options.Filename
+
+	if !checkCSVFormat(filename) {
+		return nil, fmt.Errorf("inputlookupProcessor.Process: Only .csv and .csv.gz formats are currently supported")
+	}
+
+	filePath := filepath.Join(config.GetLookupPath(), filename)
+
+	fd, err := os.Open(filePath)
+	if err != nil {
+		return nil, fmt.Errorf("inputlookupProcessor.Process: Error while opening file %v, err: %v", filePath, err)
+	}
+	defer fd.Close()
+
+	var reader *csv.Reader
+	if strings.HasSuffix(filename, ".csv.gz") {
+		gzipReader, err := gzip.NewReader(fd)
+		if err != nil {
+			return nil, fmt.Errorf("inputlookupProcessor.Process: Error while creating gzip reader, err: %v", err)
+		}
+		defer gzipReader.Close()
+		reader = csv.NewReader(gzipReader)
+	} else {
+		reader = csv.NewReader(fd)
+	}
+
+	// read columns from first row of csv file
+	columnNames, err := reader.Read()
+	if err != nil {
+		return nil, fmt.Errorf("inputlookupProcessor.Process: Error reading column names, err: %v", err)
+	}
+
+	curr := uint64(0)
+	for curr < p.start {
+		_, err := reader.Read()
```

Comment:
I think you're skipping to the `p.start` in the file in case you previously read only part of the file in a previous Process(); is that right? Is it possible to directly skip to the right position in the file using something like File.Seek()? If not, I think the code would be a little more clear if you move this logic to a helper function like `skipToStartPos()`

Reply:
addressed in the latest commit. Will check for the optimization later in next PR.